stew/objects.nim

```diff
@@ -64,3 +67,43 @@ proc baseType*(obj: RootObj): cstring =
 proc baseType*(obj: ref RootObj): cstring =
   obj[].baseType
 
+when false:
+  # TODO: Implementing this doesn't seem possible at the moment.
+  #
+  # When given enum like:
+  #
+  # type WithoutHoles2 = enum
+  #   A2 = 2, B2 = 3, C2 = 4
+  #
+  # ...the code below will print:
+  #
+  #  EnumTy
+  #    Empty
+  #    Sym "A2"
+  #    Sym "B2"
+  #    Sym "C2"
+  #
+  macro hasHoles*(T: type[enum]): bool =
+    let t = getType(T)[1]
+    echo t.treeRepr
+    return newLit(true)
+
+func checkedEnumAssign*[E: enum, I: SomeInteger](res: var E, value: I): bool =
```

Comment:
meh, what's the point of adding this when it has a pretty significant gotcha? too easy to use the wrong way

Reply:
We'll need this for any serialization format that supports enums.

Reply:
yes, but the caller won't even know their code doesn't work until it raises a defect in some edge case, which defeats the purpose of the helper - better have them write the code manually and think about their enums - or make this function warn somehow about gap support at compile time